python/cuml/cuml/metrics/regression.py

```diff
@@ -0,0 +1,368 @@
+#
+# Copyright (c) 2019-2025, NVIDIA CORPORATION.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+#
+
+from cuml.internals.safe_imports import cpu_only_import, gpu_only_import
+
+import cuml.internals
+from cuml.internals.input_utils import input_to_cupy_array
+
+np = cpu_only_import("numpy")
+cp = gpu_only_import("cupy")
+
+
+def _normalize_regression_metric_args(
+    y_true, y_pred, sample_weight, multioutput
+):
+    """
+    Helper function to normalize inputs to a regression metric.
+
+    Validates inputs and coerces all arrays to cupy of proper shape and dtype.
+    """
+    # Coerce inputs to cupy arrays
+    float_or_int = [np.float32, np.float64, np.int32, np.int64]
+    y_true, n_rows, n_cols, _ = input_to_cupy_array(
+        y_true, check_dtype=float_or_int
+    )
+    y_pred, _, _, _ = input_to_cupy_array(
+        y_pred, check_dtype=float_or_int, check_rows=n_rows, check_cols=n_cols
+    )
+    if sample_weight is not None:
+        sample_weight, _, _, _ = input_to_cupy_array(
+            sample_weight,
+            check_dtype=float_or_int,
+            check_rows=n_rows,
+            check_cols=1,
+        )
+
+    # Ensure y_true & y_pred are 2D and sample_weight is 1D
+    if y_true.ndim == 1:
+        y_true = y_true.reshape((-1, 1))
+
+    if y_pred.ndim == 1:
+        y_pred = y_pred.reshape((-1, 1))
+
+    if sample_weight is not None:
+        sample_weight = sample_weight.reshape(-1)
+
+    # Validate multioutput, and maybe coerce to a cupy array
+    valid_multioutput = ("raw_values", "uniform_average", "variance_weighted")
+    if isinstance(multioutput, str):
+        if multioutput not in valid_multioutput:
+            raise ValueError(
+                f"Valid `multioutput` values are {valid_multioutput}, got {multioutput=}"
+            )
+    elif multioutput is not None:
+        if n_cols == 1:
+            raise ValueError(
```

Comment:
Does this need to be an error or should it just be a warning?

Reply:
This is a straight port of what sklearn does. They error, so we do too.

Reply:
ah neat, was wondering what they were doing but didn't want to check the code for it :)